code/code/misc/tweaks.cc

```diff
@@ -0,0 +1,268 @@
+      if (sneezy.fetchRow()) {
+        tweak->id = (int) convertTo<int>(sneezy["tid"]);
+      }
+    }
+    else if (is_abbrev(opt, "history")) {
+      b->sendTo(COLOR_BASIC, format("\n\r<c>%s History<1>\n\r") % getTweakName(t));
+      b->sendTo(COLOR_BASIC, "<c>-----------------------------------------------------------------------------<1>\n\r");
+      b->sendTo(COLOR_BASIC, format("<c>%-15f | %-12f | %-12f | %-24s <1>\n\r")
+                                    % "Value" % "Target" % "Change Rate" % "Date");
+      sneezy.query("select tweak_id, tweak_type, tweak_value, tweak_target, tweak_rate, datecreated from globaltweaks where tweak_type = %i", t);
+      while(sneezy.fetchRow()){
+        double val=(double) convertTo<double>(sneezy["tweak_value"]);
+        double tar=(double) convertTo<double>(sneezy["tweak_target"]);
+        double rate=(double) convertTo<double>(sneezy["tweak_rate"]);
+        b->sendTo(COLOR_BASIC, format("%-15f | %-12f | %-12f | %-24s\n\r")
+                                    % val % tar % rate % sneezy["datecreated"]);
+      }
+      b->sendTo(COLOR_BASIC, "<c>-----------------------------------------------------------------------------<1>\n\r");
+    }
+    else {
+      showTweakOptions(b);
+    }
+  }
+
+  //multiple arguements given after type
+  else if ( is_abbrev(opt, "set") ){
+    sstring usage = "Usage: info tweak <parameter> set <target> [transition time in seconds]\r\n";
+
+    if (!arg || !*arg) {
+      b->sendTo(usage);
+      return;
+    } 
+
+    arg = one_argument(arg,opt, cElements(opt));
+    if (!is_number(opt) || convertTo<double>(opt)<0) {
+      b->sendTo("Invalid value given for global potential. Must be a number >= 0.0.\r\n "+usage);
+      return;
+    }
+      
+    tweak->tvalue = convertTo<double>(opt);
+
+    if (!arg || !*arg){
+      // No change period given-just set it immediately. We're done.
+      tweak->cvalue = tweak->tvalue;
+      vlogf(LOG_MISC, format("%s set %s to %f") % b->getName() % getTweakName(t) % tweak->cvalue);
+      b->sendTo(format("Target global %s set to %f\r\n") % getTweakName(t) % tweak->tvalue);
+      sneezy.query("insert into globaltweaks (tweak_type, tweak_value, tweak_target, tweak_rate) values (%i,%f,%f,0.0)", 
+        t, tweak->cvalue, tweak->tvalue, 0);
+      sneezy.query("select max( tweak_id ) as tid from globaltweaks");
+      if (sneezy.fetchRow()) {
+        tweak->id = (int) convertTo<int>(sneezy["tid"]);
+      }
+      return;
+    }
+
+    arg = one_argument(arg,opt, cElements(opt));
+    if (!is_number(opt) || convertTo<double>(opt)<=0) {
+      tweak->tvalue = tweak->cvalue;   //Restore the old target global load potential. This one was a wash.
+      b->sendTo("Invalid value given for transition time. Must be a positive number of seconds.\r\n "+usage);
+      return;
+    }
+
+    int change_period_seconds = convertTo<int>(opt);
+    tweak->rate = fabs((tweak->tvalue - tweak->cvalue)/change_period_seconds);
```

Comment:
We disagreed with Juno about whether seconds is the right unit to choose, but she wrote the code so seconds it is